#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_root_allowed.dpatch by Thomas Frauendorfer <Thomas.Frauendorfer@googlemail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Hotfix: For xmms2 0.7 .

@DPATCH@
diff --git a/src/lxmusic.c b/src/lxmusic.c
index fdc7f99..022f175 100644
--- a/src/lxmusic.c
+++ b/src/lxmusic.c
@@ -1080,7 +1080,11 @@ void on_progress_bar_changed(GtkScale* bar, gpointer user_data)
     xmmsc_result_t* res;
     gdouble p = gtk_range_get_value(GTK_RANGE(bar));
     uint32_t new_play_time = p * cur_track_duration / 100;
+#if (XMMS_IPC_PROTOCOL_VERSION >= 18)
+    res = xmmsc_playback_seek_ms( con, new_play_time, XMMS_PLAYBACK_SEEK_SET );
+#else
     res = xmmsc_playback_seek_ms( con, new_play_time );
+#endif
     xmmsc_result_unref(res);
 }

